crates/walrus-core/src/test_utils.rs

```diff
@@ -38,6 +47,135 @@ use crate::{
     },
 };
 
+/// A struct containing the test data for Quilt.
+#[derive(Debug, Clone)]
+pub struct QuiltTestData<'a> {
+    /// A map of QuiltStoreBlobs, keyed by their identifiers.
+    pub quilt_store_blobs: Vec<QuiltStoreBlob<'a>>,
+    /// A map of blob identifiers, keyed by tag keys and then tag values.
+    pub blob_identifiers_by_tag: HashMap<String, HashMap<String, HashSet<String>>>,
+}
+
+impl<'a> QuiltTestData<'a> {
+    /// Generates a new QuiltTestData from existing data.
+    pub fn new(
+        blob_data: &'a [&'a [u8]],
+        max_value_length: usize,
+        max_num_tags: usize,
+    ) -> Result<Self, QuiltError> {
+        let cow_data: Vec<Cow<'a, [u8]>> =
+            blob_data.iter().map(|&data| Cow::Borrowed(data)).collect();
+
+        let quilt_store_blobs_vec =
+            Self::generate_random_quilt_store_blobs(cow_data, max_value_length, max_num_tags)?;
+
+        Self::from_quilt_store_blobs(quilt_store_blobs_vec)
+    }
+
+    /// Generates a new QuiltTestData with new owned random data.
+    pub fn new_owned(
+        num_blobs: usize,
+        min_blob_size: usize,
+        max_blob_size: usize,
+        max_value_length: usize,
```

Comment:
Nit: maybe add doc to explain the input? IIUC, the blob is the entire walrus blob, not a quilt blob, correct?